std/range/package.d

```diff
@@ -1574,18 +1574,20 @@ Returns:
     alias of that range's type.
  */
 auto chooseAmong(Ranges...)(size_t index, Ranges rs)
-if (Ranges.length > 2
-        && is(typeof(choose(true, rs[0], rs[1])))
-        && is(typeof(chooseAmong(0, rs[1 .. $]))))
+if (Ranges.length >= 2 && allSatisfy!(isInputRange, staticMap!(Unqual, Ranges)))
```

Comment:
Because of the `is(typeof(choose(true, rs[0], rs[1])))` constraint, which reduces to the following:

```d
auto choose(R1, R2)(bool condition, R1 r1, R2 r2)
if (isInputRange!(Unqual!R1) && isInputRange!(Unqual!R2) &&
    !is(CommonType!(ElementType!(Unqual!R1), ElementType!(Unqual!R2)) == void))
```

(The common type is handles latter as `static assert` for user-friendliness)